modules/core/src/main/scala/fs2/kafka/ConsumerSettings.scala

```diff
@@ -40,12 +40,12 @@ sealed abstract class ConsumerSettings[F[_], K, V] {
   /**
     * The `Deserializer` to use for deserializing record keys.
     */
-  def keyDeserializer: F[Deserializer[F, K]]
+  def keyDeserializer: F[KeyDeserializer[F, K]]
```

Comment:
We should probably switch to allocating these in `Resource` - it allows calling `close` when wrapping Java serdes, allows widening from `Deserializer` to `KeyDeserializer`/`ValueDeserializer` without needing `Functor`, and better conveys why we're wrapping things in an effect.